snuba/cli/multistorage_consumer.py

```diff
@@ -289,30 +255,93 @@ def handler(signum: int, frame: Any) -> None:
     processor.run()
 
 
-def build_multistorage_streaming_strategy_factory(
+@dataclass
```

Comment:
We generally freeze these if they are not supposed to be mutated
```suggestion
@dataclass(frozen=True)
```